src/hooks/queryParamsHooks/usePageQueryParam.ts

```diff
@@ -0,0 +1,31 @@
+import { useValidatedQueryParam, useQueryParams, useCallback } from '@/hooks/hooks';
+import { O } from '@mobily/ts-belt';
+import { QUERY_PARAM_KEYS, INITIAL_QUERY_PARAMS_VALUE } from '@/lib/constants/constants';
+
+const usePageQueryParam = () => {
+  const { getIntParam, set } = useQueryParams();
+
+  const rawPage = useValidatedQueryParam({
+    option: getIntParam(QUERY_PARAM_KEYS.page),
+    validator: (page) => page >= INITIAL_QUERY_PARAMS_VALUE.page,
+    key: QUERY_PARAM_KEYS.page,
+    defaultValue: INITIAL_QUERY_PARAMS_VALUE.page,
+  });
+
+  const page = O.getWithDefault(rawPage, INITIAL_QUERY_PARAMS_VALUE.page);
```

Comment:
maybe it's better to return from `useValidatedQueryParam` default value instead of `O.None` instead of having here additional `O.getWithDefault`